### ISSUE TITLE
Plugin support

### ISSUE THREAD

**Issue by tj** [apex/up#453]
I prefer to have an out-of-the-box experience, however... we could support plugins in the form of executables which manipulate the Cloudformation template.

For example say you have `"plugins": ["up-rds"]` it could read RDS config from `up.json`, or you could do `["up-rds --some params --here"]` (whatever you want), and Up invokes these when planning the stack.

This would allow Up to provide more experimental support for databases etc. This only works for stack resources, maybe RPC for other styles of plugins would be ok, but IMO that gets too complex.

**Comment by olivoil** [comment]
Neat!

Do you see it limited to resource centric plugins like `up-rds`, `up-dynamo`, etc. or could there be something more general like a plugin to include any custom CF template as a nested stack and execute it via up's stack management tooling for instance?

**Comment by tj** [comment]
@olivoil yeah I think the "plugin" would basically just read from `up.json` (whatever it wants, maybe a `.dynamo` property or similar, and output to stdout some CloudFormation template. Something like that hahah.

**Comment by olivoil** [comment]
Simple enough! Could the outputs of the CF template a plugin renders be used in environment variables, maybe whitelisted via a special syntax in `up.json`? For instance if you add an `rds` plugin, it'll be handy to expose the database url via an env variable.

**Comment by tj** [comment]
@olivoil ahh good idea! I'll keep that in mind
